x/erc20/keeper/ibc_callbacks_test.go

```diff
@@ -2,574 +2,482 @@ package keeper_test
-			},
-			ackSuccess:    true,
-			receiver:      secpAddr,
-			expErc20s:     big.NewInt(0),
-			expCoins:      coins,
-			checkBalances: true,
+			name: "no-op - sender is module account",
+			transferBytes: newTransferBytes(
+				transfertypes.NewFungibleTokenPacketData(
+					unitNetwork.GetDenom(),
+					amountValue,
+					unitNetwork.App.AccountKeeper.GetModuleAddress("erc20").String(),
+					keyring.GetAccAddr(0).String(),
+					"",
+				),
+			),
+			expectedError: false,
 		},
+		// If transfer coin is evm denom there is no need for any precompile deploy or conversion
+		// Since it arrived via IBC aevmos on the other chain will look like transfer/channel-293/aevmos
+		// we will decode that and since their destination is == to our source
+		// is a native coin
 		{
-			name: "no-op - base denomination",
-			malleate: func() {
-				// base denom should be prefixed
-				sourcePrefix := transfertypes.GetDenomPrefix(transfertypes.PortID, sourceChannel)
-				prefixedDenom := sourcePrefix + s.app.StakingKeeper.BondDenom(suite.ctx)
-				transfer := transfertypes.NewFungibleTokenPacketData(prefixedDenom, "100", secpAddrCosmos, ethsecpAddrEvmos, "")
-				bz := transfertypes.ModuleCdc.MustMarshalJSON(&transfer)
-				packet = channeltypes.NewPacket(bz, 1, transfertypes.PortID, sourceChannel, transfertypes.PortID, evmosChannel, timeoutHeight, 0)
-			},
-			ackSuccess:    true,
-			receiver:      ethsecpAddr,
-			expErc20s:     big.NewInt(0),
-			expCoins:      coins,
-			checkBalances: true,
+			name: "no-op - evm Denom",
+			transferBytes: newTransferBytes(
+				transfertypes.NewFungibleTokenPacketData(
+					transfertypes.GetDenomPrefix(transfertypes.PortID, sourceChannel)+"aevmos",
+					amountValue,
+					keyring.GetAccAddr(0).String(),
+					keyring.GetAccAddr(1).String(),
+					"",
+				),
+			),
+			expectedError: false,
 		},
+		// If erc20 module is disabled its not possible to deploy any precompile
+		// or convert any coin
 		{
-			name: "no-op - pair is not registered",
-			malleate: func() {
-				transfer := transfertypes.NewFungibleTokenPacketData(erc20Denom, "100", secpAddrCosmos, ethsecpAddrEvmos, "")
-				bz := transfertypes.ModuleCdc.MustMarshalJSON(&transfer)
-				packet = channeltypes.NewPacket(bz, 1, transfertypes.PortID, sourceChannel, transfertypes.PortID, evmosChannel, timeoutHeight, 0)
-			},
-			ackSuccess:    true,
-			receiver:      ethsecpAddr,
-			expErc20s:     big.NewInt(0),
-			expCoins:      coins,
-			checkBalances: true,
+			name: "no-op - erc20 module param disabled",
+			transferBytes: newTransferBytes(
+				transfertypes.NewFungibleTokenPacketData(
+					unitNetwork.GetDenom(),
+					amountValue,
+					keyring.GetAccAddr(0).String(),
+					keyring.GetAccAddr(1).String(),
+					"",
+				),
+			),
+			expectedError: false,
 		},
+		// Is single hop and not registered Evm Extension.
+		// Should register a new EVm extension
```

Comment:
nit: prefer a consistent spelling 🙏 
```suggestion
		// Is single hop and not registered EVM Extension.
		// Should register a new EVM extension
```